hbase-backup/src/test/java/org/apache/hadoop/hbase/backup/TestIncrementalBackup.java

```diff
@@ -100,142 +141,227 @@ public void TestIncBackupRestore() throws Exception {
+    byte[] name = regions.get(0).getRegionInfo().getRegionName();
+    long startSplitTime = EnvironmentEdgeManager.currentTime();
+    try {
+      admin.splitRegionAsync(name).get();
+    } catch (Exception e) {
+      // although split fail, this may not affect following check in current API,
+      // exception will be thrown.
+      LOG.debug("region is not splittable, because " + e);
+    }
+    TEST_UTIL.waitTableAvailable(table1);
+    long endSplitTime = EnvironmentEdgeManager.currentTime();
+    // split finished
+    LOG.debug("split finished in =" + (endSplitTime - startSplitTime));
+
+    // #3 - incremental backup for multiple tables
+    tables = Lists.newArrayList(table1, table2);
+    BackupRequest request = createBackupRequest(BackupType.INCREMENTAL, tables, BACKUP_ROOT_DIR);
+    String backupIdIncMultiple = client.backupTables(request);
+    assertTrue(checkSucceeded(backupIdIncMultiple));
+    BackupManifest manifest =
+      HBackupFileSystem.getManifest(conf1, new Path(BACKUP_ROOT_DIR), backupIdIncMultiple);
+    assertEquals(Sets.newHashSet(table1, table2), new HashSet<>(manifest.getTableList()));
+    validateRootPathCanBeOverridden(BACKUP_ROOT_DIR, backupIdIncMultiple);
+
+    // add column family f2 to table1
+    // drop column family f3
+    final byte[] fam2Name = Bytes.toBytes("f2");
+    newTable1Desc = TableDescriptorBuilder.newBuilder(newTable1Desc)
+      .setColumnFamily(ColumnFamilyDescriptorBuilder.of(fam2Name)).removeColumnFamily(fam3Name)
+      .build();
+    TEST_UTIL.getAdmin().modifyTable(newTable1Desc);
+
+    // check that an incremental backup fails because the CFs don't match
+    final List<TableName> tablesCopy = tables;
+    IOException ex = assertThrows(IOException.class, () -> client
+      .backupTables(createBackupRequest(BackupType.INCREMENTAL, tablesCopy, BACKUP_ROOT_DIR)));
+    checkThrowsCFMismatch(ex, List.of(table1));
+    takeFullBackup(tables, client);
+
+    int NB_ROWS_FAM2 = 7;
+    Table t3 = insertIntoTable(conn, table1, fam2Name, 2, NB_ROWS_FAM2);
+    t3.close();
+
+    // Wait for 5 sec to make sure that old WALs were deleted
+    Thread.sleep(5000);
+
+    // #4 - additional incremental backup for multiple tables
+    request = createBackupRequest(BackupType.INCREMENTAL, tables, BACKUP_ROOT_DIR);
+    String backupIdIncMultiple2 = client.backupTables(request);
+    assertTrue(checkSucceeded(backupIdIncMultiple2));
+    validateRootPathCanBeOverridden(BACKUP_ROOT_DIR, backupIdIncMultiple2);
+
+    // #5 - restore full backup for all tables
+    TableName[] tablesRestoreFull = new TableName[] { table1, table2 };
+    TableName[] tablesMapFull = new TableName[] { table1_restore, table2_restore };
+
+    LOG.debug("Restoring full " + backupIdFull);
+    client.restore(BackupUtils.createRestoreRequest(BACKUP_ROOT_DIR, backupIdFull, false,
+      tablesRestoreFull, tablesMapFull, true));
+
+    // #6.1 - check tables for full restore
+    Admin hAdmin = TEST_UTIL.getAdmin();
```

Comment:
The Javadoc on this method says that it is managed for you. Reading the code, I'm a bit surprised that closing it doesn't result errors later on.